File: src/main/java/dev/budget/reconciler/es/ESIndex.java
package dev.budget.reconciler.es;

public enum ESIndex {
	MINT("mint", "transaction", "es/mint_index.json");

	public final String name;
	public final String type;
	public final String fileName;

	ESIndex(String name, String type, String fileName) {
		this.name = name;
		this.type = type;
		this.fileName = fileName;
	}
}

File: src/main/scala/dev/budget/reconciler/es/ESIndex.scala
package dev.budget.reconciler.es

sealed case class ESIndex(
  name: String,
  esType: String,
  indexFileName: String)

object MintESIndex extends ESIndex("mint", "transaction", "es/mint_index.json")
object YnabESIndex extends ESIndex("ynab", "transaction", "es/ynab_index.json")
